.github/workflows/build.yml

```diff
@@ -0,0 +1,66 @@
+---
+name: "node-opendds Build & Test"
+on:
+  push:
+    branches:
+      - master
+      - branch-DDS-3.*
+  pull_request:
+    branches:
+      - master
+jobs:
+  Build:
+    env:
+      - ACE_ROOT=$BUILD_DIR/ACE_TAO/ACE
+      - TAO_ROOT=$BUILD_DIR/ACE_TAO/TAO
+      - DDS_ROOT=$BUILD_DIR/OpenDDS
+      - MPC_ROOT=$BUILD_DIR/MPC
+      - LD_LIBRARY_PATH=$ACE_ROOT/lib:$DDS_ROOT/lib:$LD_LIBRARY_PATH
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version:
+          - 14.x
+        include:
+          - {os: osx, env: 'COMPILER=clang++ NO_INLINE=1 OPENDDS_BRANCH=master DDS_SECURITY=0'}
+          - {os: osx, env: 'COMPILER=clang++ NO_INLINE=1 OPENDDS_BRANCH=latest-release DDS_SECURITY=0'}
+          - {os: linux, env: 'OPENDDS_BRANCH=master DDS_SECURITY=1'}
+          - {os: linux, env: 'OPENDDS_BRANCH=latest-release DDS_SECURITY=1'}
+    runs-on: ${{ matrix.os }}
+    steps:
+      - if: runner.os == 'Linux'
+        name: Install dependencies (Ubuntu)
+        run: |-
+          sudo apt-get update
+          sudo apt-get install -y libssl-dev libxerces-c-dev
+      - name: 'Set up Node.js ${{ matrix.node-version }}'
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - uses: actions/checkout@v2
+      - run: nvm install ${{ matrix.node-version }}
+      - run: nvm use ${{ matrix.node-version }}
+      - run: git clone --depth 1 git://github.com/DOCGroup/MPC.git
+      - run: git clone --depth 1 -b Latest_Micro git://github.com/DOCGroup/ACE_TAO.git
+      - run: git clone --depth 1 -b $OPENDDS_BRANCH git://github.com/objectcomputing/OpenDDS.git
+      - run: export
+      - run: cd OpenDDS
+      - run: if [ "$DDS_SECURITY" == 1 ]; then CONFIG_OPTIONS+="--security"; fi
+      - run: ./configure --no-tests --std=c++11 $CONFIG_OPTIONS
+      - run: cd ..
+      - run: export npm_config_devdir=$BUILD_DIR/opendds-node-gyp-devdir
+      - run: npm install -g node-gyp
+      - run: node-gyp install --ensure
+      - run: if [ "$DDS_SECURITY" == 1 ]; then BUILD_TARGETS+="OpenDDS_Security"; fi
+      - run: make -C OpenDDS -sj6 DCPSInfoRepo_Main OpenDDS_Rtps_Udp $BUILD_TARGETS
+      - run: npm install
+      - run: export V8_ROOT=$npm_config_devdir/$(node -v | cut -c 2-)
+      - run: export NAN_ROOT=$BUILD_DIR/node_modules/nan
+      - run: $ACE_ROOT/bin/mwc.pl -type gnuace -exclude ACE_TAO,OpenDDS
+      - run: export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$BUILD_DIR/tools/v8stubs
+      - run: make
+      - run: if [ "$DDS_SECURITY" == 1 ]; then TEST_OPTIONS+="--secure"; fi
+      - run: cd test;
+      - run: ./run_test.pl cpp2node $TEST_OPTIONS
+      - run: ./run_test.pl node2cpp $TEST_OPTIONS
+      - run: ./run_test.pl node2node $TEST_OPTIONS
```

Comment:
Replace this with the multi line YAML syntax. I'm not even sure this is going to work correctly as is.